subprojects/docs/src/docs/userguide/dependency_management_attribute_based_matching.adoc

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[attribute_based_matching]]
+= Attribute based matching in dependency resolution
```

Comment:
There is already tons of info in there, great job!

However I find the way the concepts are introduced a bit too abstract at times. I wonder if it would not be better by starting from explaining how you could replace `latest.<status>` by attribute based matching:
* Declaring the component metadata rules to set the `status` value based on version scheme,
* Enforcing a minimal status in a configuration,
* Having exceptions to the above rule (even though that is quite verbose currently)

The above example allows to not dive immediately into exposed attributes for configurations, it focuses on the dependency level first.

And then maybe add another example that explains how to add a test fixture consumable configuration, which would allow to explain the remaining concepts.

Reply:
I wouldn't introduce using `latest.<status>` because as you know we don't have a selector for `latest`. Also it would require the introduction of the concept of variant sooner than it is there. I kind of like the soft intro to variants here.